File: stone_processor/src/main/java/com/github/klee0kai/stone/helpers/wrap/FormatInList.java
package com.github.klee0kai.stone.helpers.wrap;

import com.squareup.javapoet.CodeBlock;
import com.squareup.javapoet.TypeName;

public interface FormatInList {

    /**
     * @param originalListCode original list code
     * @param itemTransformFun format each type
     * @return
     */
    CodeBlock formatCode(TypeName originalListType, CodeBlock originalListCode, FormatSimple itemTransformFun);

}

File: stone_ksp/src/commonMain/kotlin/com/github/klee0kai/thekey/stone/ksp/helpers/wrap/FormatInList.kt
package com.github.klee0kai.thekey.stone.ksp.helpers.wrap

import com.squareup.kotlinpoet.CodeBlock
import com.squareup.kotlinpoet.TypeName

fun interface FormatInList {

    /**
     * @param itemTransformFun format each type
     * @return
     */
    fun formatCode(
        originalListType: TypeName,
        or: CodeBlock,
        itemTransformFun: FormatSimple,
    ): CodeBlock

}
